IQSS/dataverse-frontend
# 149 - Configure manual pagination in the Files Table UI

## What this PR does / why we need it:
This PR changes the configuration of the Files Table UI to use manual pagination instead of the automatic pagination provided by react-table. We need this change because we decided to apply server-side pagination to the Files Table so the API will retrieve the data by pages instead of retrieving the whole list of files at once.

## Which issue(s) this PR closes:

- Closes #149

## Special notes for your reviewer:
Changes applied:

1. getFilesByDatasetPersistentId use case modified to accept a paginationInfo argument that indicated the number of the requested page and the size of the page
2. TablePagination component now is FilesPagination component and instead of using the react table methods provided for the pagination now is using some custom methods.
3. useFilesTable hook modified to use manualPagination
4. useRowSelection hook added to manage the row selection for the is server-side pagination. There is a [bug in the react-table library](https://github.com/TanStack/table/issues/4781) when manualPagination is selected and you try to use the row selection so I had to create an hybrid between a customized row selection  feature and the row selection provided by react table.
5. RowSelectionMessage refactored to take out the logic of ClearSelection and SelectAll and move it to the useRowSelection hook

## Suggestions on how to test this:
1. `npm i`
2. `cd packages/design-system && npm run build`
3. `cd ../../`
4. `npm run storybook`
5. Check the files table at localhost:6006 storybook in the Dataset page
6. Test the pagination is working as usual going from 1 page to another.
7. Test the row selection going through the pages selecting different rows
8. Select a few rows and check that a message appears informing about the selection with 2 buttons to select All or to clear selection, ensure this 2 buttons work
9. Select a few rows with a total file size greater than 500 bytes (mocked value) to see the message about the zip download limit

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
No

## Is there a release notes update needed for this change?:
Applied manual pagination to the Files Table UI

## Additional documentation:
